### DISCUSSION TITLE
強化刪除扇區功能

### DISCUSSION THREAD

**Discussion by seantw1974** [filecoin-project/venus#4767] [Ideas]
目前扇區刪除分為Terminate(己上鏈),及Remove(未上鏈)兩種,但使用者往往會"錯誤操作"
建議統一合併為Remove, 在刪除前程式可以自己查詢扇區是否己上鏈, 再從內部執行原生的 Terminate/Remove
另外多種狀況下的扇區是無法被刪區,會一直卡住, 目前己知State 有 Packing(在Packing過程斷線,或是在/var/tmp裡沒放pieces檔造成), Deal封裝過程Packing失敗(之前是因為16G/32G扇問題), Final(ComittWait已執行,但因系統重裝或其它問題,造成檔案無法Final至存儲碟),針對這些"問題扇" 應該要有更佳的解決方式, 因此強烈建議要有 Force Delete的功能

**Comment by Fatman13** [comment]
venus暂时继承了lotus删除扇区的方法。产品会进一步讨论删除扇区的优化方式。当前的话，可以参考[这个文档](https://github.com/filecoin-project/lotus/discussions/6950)，来确保删除扇区。

**Comment by seantw1974** [comment]
這篇文章只是基礎操作，大概只有適用90%狀況，在一直刪除不了情況下，曾經使用update-state去不斷的修正，有些扇區根本不會回應update-state指令，最後只能直接動SQLite檔案，最後再確認盤的鏈上狀態.…。

Yu ***@***.***>於 2022年2月16日 週三，下午5:32寫道：

**Comment by Fatman13** [comment]
@diwufeiwen moulijian大神，这个有办法优化吗？针对@sean提出的这个问题？

**Reply by diwufeiwen** [reply]
terminate和remove慢貌似是经过了状态机，如果扇区正在进行某个任务时会等待完成，应该可以优化吧
